src/canmatrix/compare.py

```diff
@@ -20,181 +21,195 @@
 # OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
 # DAMAGE.
 
-from __future__ import print_function
 from __future__ import absolute_import
+from __future__ import print_function
+
 import logging
+import optparse
 import sys
 import typing
 
-from .log import setup_logger, set_log_level
 import canmatrix.canmatrix as cm
+import canmatrix.log as log  # import alias due to lazy import in main()
```

Comment:
No, it is not possible. The lazy import in the `main()` effectively creates local `canmatrix` variable, which overrides the global one. The `canmatrix.log` becomes "uninitialized local variable", therefore the alias and comment. We might try `global` keyword but it will be more crypric than this.

The question is rather, if the lazy init and log suppression is really needed.

Reply:
Looks like this is only used in `main()` as well and could be moved inside.  Sure, I prefer imports at the top but I think in this (not great) scenario it makes sense to have a second 'local' import rather than an aliased import further away from the usage.